src/frontend/js/confirmation-popups.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ * This file contains all functions which are needed for toggling the confirmation popup
+ */
+type EventHandler = (event: Event) => any;
+const handlers = new Map<HTMLElement, EventHandler>();
+
+window.addEventListener("load", () => {
+  // event handler for showing confirmation popups
+  document
+    .querySelectorAll(".confirmation-button")
+    .forEach((el) => el.addEventListener("click", showConfirmationPopup));
+  // event handler for closing confirmation popups
+  document
+    .querySelectorAll("#confirmation-dialog button")
+    .forEach((el) => el.addEventListener("click", closeConfirmationPopup));
+});
+
+export function showConfirmationPopup(event: Event) {
+  event.preventDefault();
+  const button = (event.target as HTMLElement).closest("button");
+  // Set confirmation data
+  document.getElementById(
+    "confirmation-subject"
+  ).innerHTML = button.getAttribute("data-confirmation-subject");
```

Comment:
Output is probably trusted because it comes from a localization file so not an issue that could be used by an attacker I guess. Still textContent makes more sense 👍🏽